.travis.yml

```diff
@@ -220,3 +220,56 @@ jobs:
       script:
       # run illuminatio e2e tests
       - python setup.py test --addopts="-m e2e"
+
+    - stage: build
+      name: "build image"
+      if: (tag IS blank)
+      sudo: required
+      services:
+        - docker
+      script:
+      - docker build -t "${TRAVIS_REPO_SLUG}"-runner:latest -f illuminatio-runner.dockerfile .
+
+    - stage: build
+      name: "push latest image"
+      if: (branch = master AND NOT type = pull_request)
+      sudo: required
+      services:
+        - docker
+      script:
+      - docker build -t "${TRAVIS_REPO_SLUG}"-runner:latest -f illuminatio-runner.dockerfile .
+      - echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+      - docker push "${TRAVIS_REPO_SLUG}"-runner:latest
+
+    - stage: build
+      name: "push tagged image"
+      if: (tag IS present)
+      sudo: required
+      services:
+        - docker
+      script:
+      - docker build -t "${TRAVIS_REPO_SLUG}"-runner:"${TRAVIS_TAG}" -f illuminatio-runner.dockerfile .
+      - echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+      - docker push "${TRAVIS_REPO_SLUG}"-runner:"${TRAVIS_TAG}"
+      - docker tag "${TRAVIS_REPO_SLUG}"-runner:"${TRAVIS_TAG}" "${TRAVIS_REPO_SLUG}"-runner:latest
```

Comment:
Why do we need to push the `latest` image here. If I create a new tag i would assume that only thee image for the according tat is created and not the `latest` image tag. IMHO `latest` should always represent the state of the current master branch.

Reply:
Right, tags could also be set to past commits so treating them as latest makes no sense.